feat: support multiple themes

sellout opened the pull request:
Like `custom-enabled-themes`, Auto-Dark now allows multiple themes to be set for either each mode. It also now defaults to using `custom-enabled-themes` if there is nothing explicitly set for Auto-Dark.

Because (dis|en)abling themes modifies `custom-enabled-themes`, you either have to set both the dark-themes and light-themes or neither, so there is a single new variable `auto-dark-themes` that manages both lists.

If neither the new variable nor `custom-enabled-themes` is set, then it falls back to the pre-existing single theme behavior, which is now marked obsolete.

With this change[^1] I can now configure my themes like
```elisp
(use-package emacs
  :custom (custom-enabled-themes '(bringhurst solarized inheritance)))

(use-package auto-dark
  :init (auto-dark-mode))
```
(since my themes all support dark/light modes).

[^1]: This change is built on https://github.com/LionyxML/auto-dark-emacs/pull/56, because it doesn’t make sense to use `custom-enabled-themes` with Auto-Dark unless it updates `frame-background-mode`.

LionyxML commented on README.md:
```diff
@@ -142,47 +141,42 @@ Following, a complete configuration with all settings set to its defaults:
+  (auto-dark-light-mode
+   . (lambda ()
+        ;; something to execute when light mode is detected
+        ))
```
Thanks! I did not noticed closing parens were commented here! 😶‍🌫️

LionyxML commented on auto-dark.el:
```diff
@@ -178,22 +219,45 @@ already set the theme for the current dark mode state."
     (unless (eq appearance auto-dark--last-dark-mode-state)
       (auto-dark--set-theme appearance))))
 
+(defun auto-dark--update-frame-backgrounds (appearance)
```
Really nice! 

Thanks for providing both functionalities for now.

Idn why we didn't use semver for this package in the beginning.

Maybe it's time for it? What do you think of we call this changes version 0.12.1? 

Then make-obsolete-variable would point to 0.13.0.

I was thinking about making a migration guide right now. But new docs + old functionality for now may be enough.

sellout replied:
Yeah, I’m happy to do whatever you want here. I certainly am in favor of SemVer (https://github.com/sellout/emacs-semver/ [^1]).

[^1]: Hmmm, it looks like I need to push a newer version of that library.

> What do you think of we call this changes version 0.12.1?
> Then make-obsolete-variable would point to 0.13.0.

I’m a little confused with this part. Are you suggesting that #56 gets tagged 0.12.1, and then this PR is tagged as 0.13.0? `make-obsolete-variable` should include whatever version we add the function call in, not a future version. And I think both #56 and this are probably equivalently minor version bumps?

A migration guide would be nice, but the warning when the code is byte-compiled does look like
```
example.el:1:1: Warning: ‘auto-dark-light-theme’ is an obsolete variable
    (as of 0.14); use ‘auto-dark-themes’ instead.
```
which isn’t the worst warning message I’ve seen.

LionyxML replied:
Hmm nice, never used this feature before.

Yeah, I was suggesting this one to become 0.12.1 and leave 0.13.0 to the future (when the old method would become obsolete).

I see it. Never used make-obsolete-variable before, If it needs the 'current version', let's give it to it.

I see no problems calling "this version" 0.13.0, if you could please bump the version number on the top of the file, it would be nice.

LionyxML reviewed (commented):
Thanks again. Awesome work!

I prefer to see your first commit merged first, just to keep things one step at the time.

While we make it happen, what about my suggestion below? (Also, this gives me a bit of time to test some stuff my self 😄 ).

sellout commented:
> Thanks again. Awesome work!

Thanks, really appreciate you being so on top of these.

> I prefer to see your first commit merged first, just to keep things one step at the time.

Yes, agreed. I’ll rebase this on master once #56 is merged, since it looks like it’s moving quickly.

LionyxML commented:
Thanks @sellout! Awesome work! #56  is already merged.

sellout commented:
Ok, this has been rebased. I updated the version and also improved some of the documentation of the affected variables. (Even though I rebased, I made the new changes in fresh commits so you can review them without having to re-review the rest of the work.)

LionyxML commented:
Hello there @sellout ! I'm (trying to be) back! 😄 

I think we have no pending matters regarding this PR. I'll make a 'release note post' on reddit, just in case people start complaining 'it stopped working' and merge this for good! 

Thanks again!

LionyxML reviewed (approved):
Outstanding!

LionyxML commented:
Note: https://www.reddit.com/r/emacs/comments/1fgcih9/autodark_013_support_multiple_themes/
